src/GPGMailer.php

```diff
@@ -29,316 +29,153 @@
-                $bounceAddress = "$parts[1]<$bounceAddress>";
-            }
-        } else {
-            $bounceAddress = $from;
-        }
-
-        // $headers["Sender"] 		= $from;
-        $headers["X-Mailer"]    = X_MAILER;
-        if (!isset($customheaders["X-Priority"])) {
-            $headers["X-Priority"]    = 3;
-        }
-
-        $headers = array_merge((array)$headers, (array)$customheaders);
-
-        // the carbon copy header has to be 'Cc', not 'CC' or 'cc' -- ensure this.
-        if (isset($headers['CC'])) {
-            $headers['Cc'] = $headers['CC'];
-            unset($headers['CC']);
-        }
-        if (isset($headers['cc'])) {
-            $headers['Cc'] = $headers['cc'];
-            unset($headers['cc']);
-        }
-
-        // Send the email
-        $headers = $this->processHeaders($headers);
-        $to = $this->validEmailAddr($to);
-
-        // Try it without the -f option if it fails
-        if (!$result = @mail($to, $subject, $fullBody, $headers, "-f$bounceAddress")) {
-            $result = mail($to, $subject, $fullBody, $headers);
-        }
-
-        if ($result) {
-            return array($to,$subject,$fullBody,$headers);
-        }
-
-        return false;
+        // set explicitly to null as empty string may be used as an invalid key
+        $this->signKey = $signingKey ?: null;
+        return $this;
     }
 
     /**
-     * Encrypting HTML emails does not work so this method triggers a warning and sends using sendPlain() and plaintext
-     * version of the HTML content.
+     * Set sign key passphrase if applicable
      *
-     * @param  string  $to            To address RFC 2822 format
-     * @param  string  $from          From address RFC 2822 format
-     * @param  string  $subject       Subject line for email
-     * @param  string  $plainContent  Content for email
-     * @param  boolean $attachedFiles Indicate whether files are attached
-     * @param  array   $customheaders Custom email headers
-     * @return mixed                  Array if successful or false if unsuccessful
+     * @param string|null $passphrase
+     * @return $this
      */
-    public function sendHTML($to, $from, $subject, $htmlContent, $attachedFiles = false, $customheaders = false, $plainContent = false)
+    public function setSignKeyPassphrase($passphrase = null)
     {
-
-        // HTML emails cannot be encrypted and create a number of issues, sendPlain() should be used instead
-        trigger_error('HTML email content cannot be encrypted, only the plain text component of this email will be generated.', E_USER_WARNING);
-
-        if (!$plainContent) {
-            $plainContent = Convert::xml2raw($htmlContent);
-        }
-
-        return $this->sendPlain($to, $from, $subject, $plainContent, $attachedFiles, $customheaders);
+        // set explicitly to null as empty string may be used as an invalid passphrase
+        $this->signKeyPassphrase = $passphrase ?: null;
```

Comment:
but '0' or 0 is valid passphrase and you will rewrite them into null in this contition

personally, i would remove the comment line and do
` $this->signKeyPassphrase = $passphrase === '' ? null : $passphrase;